spring-cloud-dataflow-server-core/src/main/java/org/springframework/cloud/dataflow/server/controller/TaskDefinitionController.java

```diff
@@ -113,9 +131,50 @@ public TaskDefinitionResource save(@RequestParam("name") String name,
 	}
 
 	/**
+	 * Register a composed task definition for future execution.
+	 *
+	 * @param name of the composed task
+	 * @param graph DSL definition for the composed task
+	 */
+	@RequestMapping(value = "/compose", method = RequestMethod.POST)
+	public TaskDefinitionResource saveComposedTask(@RequestParam("name") String name,
+			@RequestParam("definition") String graph) {
+		TaskDefinition taskDefinition = new TaskDefinition(name, graph, true);
+		TaskDefinition composedTaskDefinition = new TaskDefinition(
+				taskDefinition.getName(),getComposedDSL(taskDefinition.getDslText()));
+		validateTaskDefinitionExist(name, graph);
+		repository.save(composedTaskDefinition);
+		return taskAssembler.toResource(taskDefinition);
+	}
+
+	private void validateTaskDefinitionExist(String name, String graph) {
+		ComposedTaskValidator composedTaskValidator =
+				new ComposedTaskValidator(repository);
+		ComposedTaskNode composedTaskNode = new ComposedTaskParser().parse(name, graph);
+		composedTaskNode.accept(composedTaskValidator);
+		if(composedTaskValidator.getInvalidDefinitions().size() > 0) {
+			List<ComposedTaskValidationProblem> composedTaskValidationProblems =
+					new ArrayList<>();
+			for(String taskName : composedTaskValidator.getInvalidDefinitions().keySet()) {
```

Comment:
Your crystal ball is incredibly accurate.  Does it pick winning lottery numbers ;-)